src/pages/register-a-beacon/beacon-owner-address.tsx

```diff
@@ -0,0 +1,157 @@
+import { GetServerSideProps } from "next";
+import React, { FunctionComponent } from "react";
+import { BackButton, Button } from "../../components/Button";
+import { FormErrorSummary } from "../../components/ErrorSummary";
+import {
+  Form,
+  FormFieldset,
+  FormGroup,
+  FormLegendPageHeading,
+} from "../../components/Form";
+import { Grid } from "../../components/Grid";
+import { FormInputProps, Input } from "../../components/Input";
+import { Layout } from "../../components/Layout";
+import { IfYouNeedHelp } from "../../components/Mca";
+import { GovUKBody } from "../../components/Typography";
+import { FormValidator } from "../../lib/formValidator";
+import { FormPageProps, handlePageRequest } from "../../lib/handlePageRequest";
+import { ensureFormDataHasKeys } from "../../lib/utils";
+
+interface BuildingNumberAndStreetInputProps {
+  valueLine1: string;
+  valueLine2: string;
+  showErrors: boolean;
+  errorMessages: string[];
+}
+
+const BeaconOwnerAddressPage: FunctionComponent<FormPageProps> = ({
+  formData,
+  needsValidation,
+}: FormPageProps): JSX.Element => {
+  formData = ensureFormDataHasKeys(
+    formData,
+    "beaconOwnerAddressLine1",
+    "beaconOwnerAddressLine2",
+    "beaconOwnerTownOrCity",
+    "beaconOwnerCounty",
+    "beaconOwnerPostcode"
+  );
+  const pageHeading = "What is the beacon owner's address?";
+  const errors = FormValidator.errorSummary(formData);
+  const {
+    beaconOwnerAddressLine1,
+    beaconOwnerTownOrCity,
+    beaconOwnerPostcode,
+  } = FormValidator.validate(formData);
+  const pageHasErrors = needsValidation && FormValidator.hasErrors(formData);
+
+  return (
+    <Layout
+      navigation={<BackButton href="/register-a-beacon/about-beacon-owner" />}
+      title={pageHeading}
+      pageHasErrors={pageHasErrors}
+    >
+      <Grid
+        mainContent={
+          <>
+            <Form action="/register-a-beacon/beacon-owner-address">
+              <FormFieldset>
+                <FormErrorSummary
+                  needsValidation={needsValidation}
```

Comment:
I think renaming this variable to something like: `showErrorSummary` would be more obvious rather than `needsValidation`?